keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryResourceRepository.kt

```diff
@@ -39,6 +41,7 @@ class InMemoryResourceRepository(
   private val resourceEvents = mutableMapOf<String, MutableList<ResourceEvent>>()
   private val applicationEvents = mutableMapOf<String, MutableList<ApplicationEvent>>()
   private val lastCheckTimes = mutableMapOf<String, Instant>()
+  private val resourceArtifacts = mutableMapOf<Resource<*>, DeliveryArtifact>()
```

Comment:
probably makes more sense to have `mutableMapOf<String, DeliveryArtifact>()`, I don't think storing the whole resource here gets us anything & will make things more confusing (equality of objects is not as easy as i want it to be, I've had lots of troubles here in the past)

Reply:
I haven't had any issues, but will change it for consistency.